granary/bluesky.py

```diff
@@ -44,59 +53,53 @@ def _maybe_load_lexicons():
 def url_to_did_web(url):
   """Converts a URL to a did:web.
 
+  In AT Proto, only hostname-based web DIDs are supported.
+  Paths are not supported, and will be discarded.
+
   Examples:
   * 'https://foo.com' => 'did:web:foo.com'
-  * 'https://foo.com:3000' => 'did:web:foo.com%3A3000'
-  * 'https://bar.com/baz/baj' => 'did:web:bar.com:baz:baj'
-
-  https://w3c-ccg.github.io/did-method-web/#example-creating-the-did
+  * 'https://foo.com:3000' => 'did:web:foo.com'
+  * 'https://foo.bar.com/baz/baj' => 'did:web:foo.bar.com'
 
-  TODO: require https?
+  https://atproto.com/specs/did
 
   Args:
     url: str
 
   Returns: str
   """
   parsed = urllib.parse.urlparse(url)
-  if not parsed.netloc:
+  if not parsed.hostname:
     raise ValueError(f'Invalid URL: {url}')
 
-  did = f'did:web:{urllib.parse.quote(parsed.netloc)}'
-  if parsed.path:
-    did += f'{parsed.path.replace("/", ":")}'
-
-  return did.strip(':')
+  return f'did:web:{parsed.hostname}'
 
 
 def did_web_to_url(did):
   """Converts a did:web to a URL.
 
+  In AT Proto, only hostname-based web DIDs are supported.
+  Paths are not supported, and will throw an invalid error.
+
   Examples:
   * 'did:web:foo.com' => 'https://foo.com'
-  * 'did:web:foo.com%3A3000' => 'https://foo.com:3000'
-  * 'did:web:bar.com:baz:baj' => 'https://bar.com/baz/baj'
+  * 'did:web:foo.com%3A3000' => INVALID
+  * 'did:web:bar.com:baz:baj' => INVALID
 
-  https://w3c-ccg.github.io/did-method-web/#read-resolve
+  https://atproto.com/specs/did
 
   Args:
     did: str
 
   Returns: str
   """
-  if not did or not did.startswith('did:web:'):
+  if not did or not DID_WEB_PATTERN.search(did):
```

Comment:
minor nit: consider `match` instead, to make it clear that it has to match from the beginning? (technically your leading `^` does that anyway, but this might help avoid any confusion.)

Reply:
Cool can do, this was just due to my lack of Python knowledge